src/web/app/pages-instructor/instructor-session-individual-extension-page/instructor-session-individual-extension-page.component.ts

```diff
@@ -0,0 +1,569 @@
+    } else {
+      record = { ...this.instructorDeadlines };
+    }
+
+    selectedIndividuals.forEach((x) => {
+      record[x.email] = extensionTimestamp;
+    });
+    return record;
+  }
+
+  private handleDeleteDeadlines(
+    selectedStudents: StudentExtensionTableColumnModel[],
+    selectedInstructors: InstructorExtensionTableColumnModel[],
+    isNotifyDeadlines: boolean,
+  ): void {
+    const request: FeedbackSessionUpdateRequest = {
+      studentDeadlines: this.getDeletedDeadlines(selectedStudents, true),
+      instructorDeadlines: this.getDeletedDeadlines(selectedInstructors, false),
+      ...this.feedbackSessionDetails,
+    };
+
+    this.isSubmittingDeadlines = true;
+    this.feedbackSessionsService
+      .updateFeedbackSession(this.courseId, this.feedbackSessionName, request, isNotifyDeadlines)
+      .pipe(finalize(() => { this.isSubmittingDeadlines = false; }))
+      .subscribe(
+        () => {
+          this.loadFeedbackSessionAndIndividuals();
+          this.statusMessageService.showSuccessToast(
+            `Successfully deleleted extension(s) for ${selectedStudents.length} student(s) and`
+            + ` ${selectedInstructors.length} instructor(s)!`,
+          );
+        },
+        (resp: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorToast(resp.error.message);
+        },
+      );
+  }
+
+  private getDeletedDeadlines(
+    selectedIndividuals: StudentExtensionTableColumnModel[] | InstructorExtensionTableColumnModel[],
+    isStudent: boolean,
+  ): Record<string, number> {
+    let record: Record<string, number> = {};
+    if (isStudent) {
+      record = { ...this.studentDeadlines };
+    } else {
+      record = { ...this.instructorDeadlines };
+    }
+
+    selectedIndividuals.forEach((x) => {
+      delete record[x.email];
+    });
+    return record;
+  }
+
+  private getSelectedStudents(): StudentExtensionTableColumnModel[] {
+    return this.studentsOfCourse.filter((x) => x.isSelected);
+  }
+
+  getNumberOfSelectedStudents(): number {
+    return this.getSelectedStudents().length;
+  }
+
+  private getSelectedStudentsWithExtensions(): StudentExtensionTableColumnModel[] {
+    return this.studentsOfCourse.filter((x) => x.isSelected && x.hasExtension);
+  }
+
+  private getSelectedInstructors(): InstructorExtensionTableColumnModel[] {
+    return this.instructorsOfCourse.filter((x) => x.isSelected);
+  }
+
+  private getNumberOfSelectedInstructors(): number {
+    return this.getSelectedInstructors().length;
+  }
+
+  private getSelectedInstructorsWithExtensions(): InstructorExtensionTableColumnModel[] {
+    return this.instructorsOfCourse.filter((x) => x.isSelected && x.hasExtension);
+  }
+
+  hasSelected(): boolean {
+    for (const student of this.studentsOfCourse) {
+      if (student.isSelected) return true;
+    }
+    for (const instructor of this.instructorsOfCourse) {
+      if (instructor.isSelected) return true;
+    }
```

Comment:
Hack, but would this be simpler if we do the following:
```suggestion
    [...this.studentsOfCourse, ...this.instructorsOfCourse]
      .some(user => user.isSelected);
```